sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala

```diff
@@ -50,7 +50,7 @@ object ExpressionSet {
 class ExpressionSet protected(
     protected val baseSet: mutable.Set[Expression] = new mutable.HashSet,
     protected val originals: mutable.Buffer[Expression] = new ArrayBuffer)
-  extends Set[Expression] {
+  extends Set[Expression] with Serializable {
```

Comment:
Yes, I got an exception regarding non-serializable in test suites in ```hive``` when ```ExpressionSet``` is not ```Serializable```. This is why I added ```Serialiable``` to ```ExpressionSet```